packages/backend-defaults/src/entrypoints/actions/DefaultActionsService.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  ActionsService,
+  ActionsServiceAction,
+  AuthService,
+  BackstageCredentials,
+  DiscoveryService,
+  LoggerService,
+  RootConfigService,
+} from '@backstage/backend-plugin-api';
+import { ResponseError } from '@backstage/errors';
+import { JsonObject } from '@backstage/types';
+
+export class DefaultActionsService implements ActionsService {
+  private constructor(
+    private readonly discovery: DiscoveryService,
+    private readonly config: RootConfigService,
+    private readonly logger: LoggerService,
+    private readonly auth: AuthService,
+  ) {}
+
+  static create({
+    discovery,
+    config,
+    logger,
+    auth,
+  }: {
+    discovery: DiscoveryService;
+    config: RootConfigService;
+    logger: LoggerService;
+    auth: AuthService;
+  }) {
+    return new DefaultActionsService(discovery, config, logger, auth);
+  }
+
+  async listActions({ credentials }: { credentials: BackstageCredentials }) {
+    const pluginSources =
+      this.config.getOptionalStringArray('backend.actions.pluginSources') ?? [];
+
+    const remoteActionsList = await Promise.all(
+      pluginSources.map(async source => {
+        const pluginBaseUrl = await this.discovery.getBaseUrl(source);
+        const response = await this.makeRequest({
+          url: `${pluginBaseUrl}/.backstage/actions/v1/actions`,
+          pluginId: source,
+          credentials,
+        });
+
+        if (!response.ok) {
```

Comment:
Given this fallback I think it'd make sense to allow a failing request too, i.e. if `makeRequest` throws

Reply:
Do you mean wrap the entire thing in `try/catch` with `ForwardedError`?